### PR TITLE
feat(akamai): ✨ support adding multiple entries to client list

### PR DESCRIPTION
**Pull request by laurasfo**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description

The `akamai-add-client-list-entry` command previously only supported adding a **single entry** per API call. The Akamai Client List API (`POST /client-list/v1/lists/{listId}/items`) natively supports batch append via an array of entries, but this was not exposed in the integration.

This PR updates the command to accept a **comma-separated list of values**, allowing multiple entries to be added to a client list in a single API call.

### Changes
- **`Akamai_WAF.py`**: Updated `add_client_list_entry_command` and `Client.add_client_list_entry` to use `argToList()` for the `value` and `tags` arguments, building a batch `append` array for the API request. Added validation to ensure at least one value is provided before calling the API.
- **`Akamai_WAF.yml`**: Added `isArray: true` to the `value` argument. Updated descriptions for `value`, `description`, `expiration_date`, and `tags` to clarify batch behavior.
- **`Akamai_WAF_test.py`**: Added `test_add_client_list_entry_command_multiple_values` to cover the multi-entry case.
- **`README.md`**: Added command example and expected human-readable output.
- **`ReleaseNotes/2_4_0.md`** + **`pack_metadata.json`**: Version bumped to `2.4.0`.

### Example usage
!akamai-add-client-list-entry list_id="1234_MYIPLIST" value="1.2.3.4/32,1.2.3.5/32"

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by laurasfo** [comment]
@marketplace-ai-reviewer start review

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 1]
```diff
@@ -502,24 +502,42 @@ def get_client_list_activation_status(self, list_id: str, network_environment: s
...
         Returns:
             Json response as dictionary
         """
-        tags = tags.split(",") if tags else []
+        tags_list = tags.split(",") if tags else []
         exp_iso = normalize_to_iso8601(expiration_date) if expiration_date else None
-        entry = {"value": value, "description": description, "expirationDate": exp_iso, "tags": tags}
-        body = {"append": [entry]}
-        return self._http_request(method="POST", url_suffix=f"/client-list/v1/lists/{list_id}/items", json_data=body)
+        values = [v.strip() for v in value.split(",") if v.strip()] if value else []
```
[the remark this answers is not in the document]
Comment (laurasfo) [reply Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py]: "isArray: true" was removed

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py [hunk 2]
```diff
@@ -751,6 +751,33 @@ def test_add_client_list_entry_command(mocker, akamai_waf_client):
...
+
+    mocker.patch.object(akamai_waf_client, "add_client_list_entry", return_value=expected_raw_response)
+
+    human_readable, _, raw_response = add_client_list_entry_command(
+        client=akamai_waf_client,
+        list_id="12345",
+        value="1.2.3.4,1.2.3.5,1.2.3.6",
+        description="Test entry",
+        tags=["test_tag"],
+    )
```
[the remark this answers is not in the document]
Comment (laurasfo) [reply Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py:883]: "isArray: true" was removed

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 3]
```diff
@@ -3789,22 +3807,31 @@ def activate_client_list_command(
...
     Returns:
         Human readable, context entry, raw response
     """
     raw_response = client.add_client_list_entry(list_id, value, description, expiration_date, tags)
-    human_readable = f"Entry '{value}' added successfully to Akamai WAF Client List {list_id}."
+    values = [v.strip() for v in value.split(",") if v.strip()]
+    if len(values) == 1:
+        human_readable = f"Entry '{values[0]}' added successfully to Akamai WAF Client List {list_id}."
+    else:
+        human_readable = f"Entries '{', '.join(values)}' added successfully to Akamai WAF Client List {list_id}."
```
[the remark this answers is not in the document]
Comment (laurasfo) [reply Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py:3836]: "isArray: true" was removed

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by laurasfo** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/demisto/content/pull/44130#pullrequestreview-4205231044)

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 4]
```diff
@@ -502,24 +502,42 @@ def get_client_list_activation_status(self, list_id: str, network_environment: s
...
-            tags: A list of tags for the new entry.
+            value: A comma-separated list of values to add.
+            description: A description for the new entries. Applied to all entries.
+            expiration_date: The expiration date for the new entries. Applied to all entries.
+            tags: A comma-separated list of tags for the new entries. Applied to all entries.
         Returns:
             Json response as dictionary
         """
-        tags = tags.split(",") if tags else []
+        tags_list = tags.split(",") if tags else []
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py]: Consider using `argToList` for parsing comma-separated strings.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py (hunk 1 above)
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py]: - Consider using `argToList` to simplify this list comprehension.
- Use `argToList` for robust parsing of comma-separated strings and lists.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 5]
```diff
@@ -3789,22 +3807,31 @@ def activate_client_list_command(
...
+        value: A comma-separated list of values to add.
+        description: A description for the new entries. Applied to all entries.
+        expiration_date: The expiration date for the new entries. Applied to all entries.
+        tags: A comma-separated list of tags for the new entries. Applied to all entries.
     Returns:
         Human readable, context entry, raw response
     """
     raw_response = client.add_client_list_entry(list_id, value, description, expiration_date, tags)
-    human_readable = f"Entry '{value}' added successfully to Akamai WAF Client List {list_id}."
+    values = [v.strip() for v in value.split(",") if v.strip()]
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py]: Consider using `argToList` here for consistency.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py [hunk 6]
```diff
@@ -751,6 +751,33 @@ def test_add_client_list_entry_command(mocker, akamai_waf_client):
...
+    expected_raw_response = {}
+
+    mocker.patch.object(akamai_waf_client, "add_client_list_entry", return_value=expected_raw_response)
+
+    human_readable, _, raw_response = add_client_list_entry_command(
+        client=akamai_waf_client,
+        list_id="12345",
+        value="1.2.3.4,1.2.3.5,1.2.3.6",
+        description="Test entry",
+        tags=["test_tag"],
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py]: Pass `tags` as a string instead of a list.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py (hunk 3 above)
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py:3836]: Handle empty values and use `argToList`.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml [hunk 7]
```diff
@@ -719,21 +719,21 @@ script:
...
-    description: Adds an entry to a client list.
+    description: Adds one or more entries to a client list.
     arguments:
     - name: list_id
       required: true
       description: The ID of the client list.
     - name: value
       required: true
-      description: The value of the new entry.
+      description: A comma-separated list of values to add to the client list (e.g. 1.2.3.4/32,1.2.3.5/32).
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml:730]: Consider adding `isArray: true` to the `value` argument since it now accepts a comma-separated list of values. This allows users to pass arrays directly from context in playbooks, improving usability. This is also consistent with the `tags` argument and the `akamai-remove-client-list-entry` command.

```yaml
        isArray: true
        description: A comma-separated list of values to add to the client list (e.g. 1.2.3.4/32,1.2.3.5/32).
```

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md [hunk 8]
```diff
@@ -414,15 +414,14 @@ Adds an entry to a client list.
...
 | --- | --- | --- |
 | list_id | The ID of the client list. | Required |
-| value | The value of the new entry. | Required |
-| description | A description of the new entry. | Optional |
-| expiration_date | The expiration date for the new entry. Use ISO 8601 format (e.g. 2025-09-29 or 2025-09-29T13:15:28). | Optional |
-| tags | A list of tags for the new entry. | Optional |
+| value | A comma-separated list of values to add to the client list (e.g. 1.2.3.4/32,1.2.3.5/32). | Required |
+| description | A description for the new entries. Applied to all entries. | Optional |
+| expiration_date | The expiration date for the new entries. Applied to all entries. Use ISO 8601 format (e.g. 2025-09-29 or 2025-09-29T13:15:28). | Optional |
+| tags | A list of tags for the new entries. Applied to all entries. | Optional |
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md:421]: The command section is missing a `Command Example` section.

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Akamai WAF integration! I've reviewed your changes and have a few quick suggestions to improve consistency and ensure validation passes. 

Please fix a critical YAML validation issue by making `name` or `display` the first key, and consistently use `argToList` in the Python code for parsing comma-separated strings. Finally, remember to add a command example to the README and include 'Akamai' in the pack metadata keywords. 

Great work overall!

Additionally, please address the following file-level notes:
- `Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml`: **CRITICAL**: The `.yml` file must start with `name` or `display` as the first key.

According to XSOAR YAML Structure Validation guidelines, the first key defined at the root level of the `.yml` file must always be `name` or `display`. Currently, the file starts with `category`. Please reorder the root keys to comply with this requirement.
- `Packs/Akamai_WAF/pack_metadata.json`: The vendor name `Akamai` is missing from the `keywords` list. Please add it to ensure the pack is easily discoverable.

 @copilot-pull-request-reviewer[bot], @kamalq97, @laurasfo please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Review by marketplace-ai-reviewer — COMMENTED** [review]
#### 🔄 Re-reviewing updates...
> [!IMPORTANT]
> To maintain review integrity, threads marked as **Resolved** without accompanying changes or explanations will be automatically **unresolved** for further verification.

Thank you for your contribution! Re-reviewed your PR, No new issues found. Great work! 👍 
Please check the previous comments to ensure all feedback has been resolved.

 @kamalq97, @copilot-pull-request-reviewer[bot], @laurasfo, @marketplace-ai-reviewer please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml (hunk 7 above)
Comment (marketplace-ai-reviewer) [reply Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml:730]: This thread was marked as resolved, but I don't see the suggested change in the latest diff, nor an explanation. Could you please clarify if this will be addressed or if there's a reason to keep it as is?

**Comment by laurasfo** [comment]
@marketplace-ai-reviewer start review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 9]
```diff
@@ -3789,22 +3809,33 @@ def activate_client_list_command(
 
 @logger
 def add_client_list_entry_command(
-    client: Client, list_id: str, value: str, description: str = None, expiration_date: str = None, tags: str = None
+    client: Client,
+    list_id: str,
+    value: str,
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py:3812]: The PR description mentions adding `isArray: true` to the `value` argument in the YAML file, but it appears to be missing from `Akamai_WAF.yml` on the source branch.

Please ensure `isArray: true` is added to the `value` argument for the `akamai-add-client-list-entry` command in the YAML file so the UI correctly handles multi-select and list inputs.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py [hunk 10]
```diff
@@ -751,6 +751,33 @@ def test_add_client_list_entry_command(mocker, akamai_waf_client):
     assert expected_human_readable == human_readable
 
 
+def test_add_client_list_entry_command_multiple_values(mocker, akamai_waf_client):
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py:859]: Consider adding a test case to verify the `ValueError` exception when `value` is empty.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml (hunk 7 above)
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml:730]: The PR description mentions adding `isArray: true` to the `value` argument, but it is missing here. Please add it to ensure the platform correctly handles list inputs.

```yaml
        isArray: true
        description: A comma-separated list of values to add to the client list (e.g. 1.2.3.4/32,1.2.3.5/32).
```

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md [hunk 11]
```diff
@@ -414,15 +414,23 @@ Adds an entry to a client list.
...
 
 #### Context Output
 
 There is no context output for this command.
 
+#### Command Example
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md]: The standard header generated by the SDK is `#### Command example` (lowercase 'e'). It appears this section might have been hand-written or manually modified.

```suggestion
#### Command example
```

Additionally, please consider running `demisto-sdk generate-docs --insecure -e <path/to/command_examples.txt>` to regenerate the entire command section with proper formatting, examples, and context output, as per the [Creating Documentation](https://xsoar.pan.dev/docs/documentation/readme_file#creating-documentation) guidelines.

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Akamai WAF integration! I've reviewed your changes and left a few notes. 

The main issue to look into is that the `isArray: true` flag mentioned in your PR description is missing from the YAML file. I also left a couple of suggestions regarding test coverage for empty values and regenerating the README to fix some minor formatting issues. 

Thanks again for your work on this!

Additionally, please address the following file-level notes:
- `Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py`: Missing YAML changes in PR

 @kamalq97, @marketplace-ai-reviewer, @copilot-pull-request-reviewer[bot], @laurasfo please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by laurasfo** [comment]
@marketplace-ai-reviewer start review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py [hunk 12]
```diff
@@ -3789,22 +3809,33 @@ def activate_client_list_command(
...
-        expiration_date: The expiration date for the new entry.
-        tags: A list of tags for the new entry.
+        value: A comma-separated list of values to add.
+        description: A description for the new entries. Applied to all entries.
+        expiration_date: The expiration date for the new entries. Applied to all entries.
+        tags: A comma-separated list of tags for the new entries. Applied to all entries.
     Returns:
         Human readable, context entry, raw response
     """
+    values = argToList(value)
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py:3829]: Consider removing duplicate validation to keep the command handler cleaner.

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Akamai WAF integration! The code looks good overall, but I left a minor suggestion regarding maintainability. Consider removing the duplicate validation to keep the command handler cleaner. Let me know if you have any questions!

 @copilot-pull-request-reviewer[bot], @marketplace-ai-reviewer, @kamalq97, @laurasfo please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by laurasfo** [comment]
@marketplace-ai-reviewer start review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py [hunk 13]
```diff
@@ -742,15 +841,73 @@ def test_add_client_list_entry_command(mocker, akamai_waf_client):
...
+        list_id="12345",
+        value="1.2.3.4",
+        description="Test entry",
+        tags=["test_tag"],
     )
     assert expected_raw_response == raw_response
     assert expected_human_readable == human_readable
 
 
+def test_add_client_list_entry_command_multiple_values(mocker, akamai_waf_client):
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF_test.py:859]: Consider adding a unit test for `Client.add_client_list_entry` using `requests_mock`.

File: Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md [hunk 14]
```diff
@@ -414,15 +414,23 @@ Adds an entry to a client list.
...
 
 There is no context output for this command.
 
+#### Command example
+
+```!akamai-add-client-list-entry list_id="1234_MYIPLIST" value="1.2.3.4/32,1.2.3.5/32"```
+
+#### Human Readable Output
+
+>Entries '1.2.3.4/32, 1.2.3.5/32' added successfully to Akamai WAF Client List 1234_MYIPLIST.
```
Comment (marketplace-ai-reviewer) [inline Packs/Akamai_WAF/Integrations/Akamai_WAF/README.md]: The blockquote for the Human Readable Output is missing a space after the `>` character. In MDX, a raw `>` without a space might not be parsed as a blockquote and could cause rendering issues.

```suggestion
> Entries '1.2.3.4/32, 1.2.3.5/32' added successfully to Akamai WAF Client List 1234_MYIPLIST.
```

Additionally, as this section appears to be hand-written, we highly recommend [generating the documentation](https://xsoar.pan.dev/docs/documentation/readme_file#run-demisto-sdk-generate-docs) using the SDK to ensure consistent formatting across all commands:
`demisto-sdk generate-docs --insecure -e <path/to/command_examples.txt> -i Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml`

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thanks for your contribution to the Akamai WAF pack! I've reviewed the PR and left a few notes to help polish the integration. 

Please update the new command to return a `CommandResults` object, define its context outputs in the YML, and add a corresponding unit test. Additionally, make sure to remove the email address from the contributors file and consider auto-generating the README to resolve some minor formatting issues. 

Thanks again for your hard work!

Additionally, please address the following file-level notes:
- `Packs/Akamai_WAF/CONTRIBUTORS.json`: Please remove the email address from the CONTRIBUTORS.json file.
- `Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.py`: - The updated command `add_client_list_entry_command` should return a `CommandResults` object instead of a tuple.
- Use `arg_to_datetime` for date parsing in `normalize_to_iso8601`.
- `Packs/Akamai_WAF/Integrations/Akamai_WAF/Akamai_WAF.yml`: Define context outputs for the `akamai-add-client-list-entry` command.

 @copilot-pull-request-reviewer[bot], @marketplace-ai-reviewer, @kamalq97, @laurasfo please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by laurasfo** [comment]
@marketplace-ai-reviewer start review

**Comment by marketplace-ai-reviewer** [comment]
🤖 Analysis started. Please wait for results...

**Comment by marketplace-ai-reviewer** [comment]
<details>
<summary><strong>🤖 AI Review Disclaimer</strong></summary>

This review was generated by an AI-powered tool and may contain inaccuracies. Please be advised, and we extend our sincere apologies for any inconvenience this may cause.
</details>

**Review by marketplace-ai-reviewer — COMMENTED** [review]
Thank you for your contribution! No issues found. Great work! 👍

 @laurasfo, @copilot-pull-request-reviewer[bot], @kamalq97, @marketplace-ai-reviewer please review and approve the results generated by the AI Reviewer by responding 👍 on this comment.

**Comment by kamalq97** [comment]
Hi @laurasfo, thank you for the contribution!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me know when you're available for one over the DFIR Community Slack. 

Feel free also to send me a video recording of a demo if you prefer.

**Comment by laurasfo** [comment]
Hello @kamalq97 
Video attached: https://github.com/user-attachments/assets/fb417585-cf48-4049-b59f-015f813f80dd

**Review by kamalq97 — APPROVED** [review]
